syntax = "proto3";
package manager;

service ClientService{
    rpc RegisterClient (RegisterClientRequest) returns (Client);
    rpc ConsultClient (ConsultClientRequest) returns (Client);
}

message Client {
    string id = 1;
    string name = 2;
    string credit_limit = 3;
    double balance = 4;
}

message RegisterClientRequest {
    string name= 1;
    double credit_limit= 2;
}

message ConsultClientRequest {
    string id = 1;
}

// -------------------------------------------------------

service TransactionService {
    rpc RequestTransaction (PerformTransaction) returns (TransactionResponse);
    rpc ConsultTransaction (ConsultClientRequest) returns (stream Transaction);
}

message PerformTransaction {
    string client_id = 1;
    double value = 2;
    string type = 3;
    string description = 4;
}

message TransactionResponse {
    bool success = 1;
    string message = 2;
    double balance = 3;
}

message Transaction {
    string id = 1;
    string client_id = 2;
    double value = 3;
    string type = 4;
    string description = 5;
    string date = 6;
}